### PR TITLE
Refactor validation

### PR DESCRIPTION
**Pull request by a-cool-train**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by TerryHowe — COMMENTED** [review]
looks good other than the unused methods

File: api/v1alpha1/packagebundle.go [hunk 1]
```diff
@@ -152,6 +165,29 @@ func (s BundlePackageSource) PackageMatches(other BundlePackageSource) bool {
 	return true
 }
 
+func (bp *BundlePackage) GetJsonSchema() ([]byte, error) {
+	// The package configuration is gzipped and base64 encoded
+	// When processing the configuration, the reverse occurs: base64 decode, then unzip
+	configuration := bp.Source.Versions[0].Schema
```
Comment (TerryHowe) [inline api/v1alpha1/packagebundle.go:171]: outside of the scope of this PR, but this will obviously not work if they are installing Versions[1]

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: api/v1alpha1/packagebundle.go (hunk 1 above)
Comment (a-cool-train) [reply api/v1alpha1/packagebundle.go:171]: Currently I am not aware of any mechanism to provide the version of packages. Maybe once adding pinning to packages, we can decide to update this.

Comment (TerryHowe) [reply api/v1alpha1/packagebundle.go:171]: PackageVersion

**Pull request merged**